src/lib/storage/chunk.cpp

```diff
@@ -240,16 +242,15 @@ std::vector<std::shared_ptr<const AbstractSegment>> Chunk::_get_segments_for_ids
   return segments;
 }
 
-const std::optional<ChunkPruningStatistics>& Chunk::pruning_statistics() const {
-  return _pruning_statistics;
+std::shared_ptr<const ChunkPruningStatistics> Chunk::pruning_statistics() const {
+  return std::atomic_load(&_pruning_statistics);
```

Comment:
~We do we need the atomic load here?~

With the updated CI, maybe `std::atomic<std::shared_ptr<PruningDings>>` does now work?